[DOCS-11126] Add note about Splunk HEC token env var (#30700)
DataDog/documentation: layouts/shortcodes/observability_pipelines/configure_existing_pipelines/destination_env_vars/splunk_hec.en.md

- Splunk HEC token:

>     - The Splunk HEC token for the Splunk indexer. **Note**: Depending on your OS and shell version, you may want to avoid adding quotes to the environment variable.

Reviewer: how about this ? 
- using "environment" instead of "OS" just to be even broader, sounds like things like Docker can affect this too
-  slight change to "avoid adding quotes" because our UI is the one that adds those quotes (we included those pretty recently in an attempt to help people escape special characters)
```suggestion
    - The Splunk HEC token for the Splunk indexer. **Note**: Depending on your shell and environment, you may want to to ensure the environment variable value is not wrapped in quotes.
```
